test_server/src/main/scala/io/reactivesocket/tck/DSLTestSubscriber.scala

```diff
@@ -0,0 +1,107 @@
+    this(writer, argMap, "", "", "channel", marble)
+  }
+
+  private var id: UUID = null
+  this.id = UUID.randomUUID
+
+  // decide what type of subscriber to write down
+  if (kind.equals("")) writer.write("") // write nothing
+  else if (kind.equals("channel") && argMap == null) writer.write("subscribe%%channel%%" + this.getID + "%%"
+    + writechannel(marble) + "\n")
+  else if(argMap != null) writer.write("subscribe%%channel%%" + this.getID + "%%" + writechannel(marble) + "&&" + write(argMap)
+    + "\n")
+  else writer.write("subscribe%%" + kind + "%%" + this.getID + "%%" + initData + "%%" + initMeta + "\n")
+
+  def getID: String = return this.id.toString
+
+  override def onSubscribe(s: Subscription) : Unit =  {}
+
+  override def onNext(t: Payload) : Unit = {}
+
+  override def onError(t: Throwable) : Unit = {}
+
+  override def onComplete() : Unit = {}
+
+  override def request(n: Long) : Unit = writer.write("request%%" + n + "%%" + this.id + "\n")
+
+  override def cancel() : Unit = writer.write("cancel%%" + this.id + "\n")
+
+
+  // assertion tests
+
+  def assertNoErrors() : Unit = writer.write("assert%%no_error%%" + this.id + "\n")
+
+  def assertError() : Unit = writer.write("assert%%error%%" + this.id + "\n")
+
+  def assertReceived(lst: List[(String, String)]) : Unit =
+    writer.write("assert%%received%%" + this.id + "%%" + printList(lst) + "\n")
+
+  def assertReceivedCount(n: Long) : Unit = writer.write("assert%%received_n%%" + this.id + "%%" + n + "\n")
+
+  def assertReceivedAtLeast(n: Long) : Unit = writer.write("assert%%received_at_least%%" + this.id + "%%" + n + "\n")
+
+  def assertNoValues() : Unit = writer.write("assert%%no_values%%" + this.id + "\n")
+
+  def assertCompleted() : Unit = writer.write("assert%%completed%%" + this.id + "\n")
+
+  def assertNotCompleted() : Unit = writer.write("assert%%no_completed%%" + this.id + "\n")
+
+  def assertNotTerminated() : Unit = writer.write("assert%%no_terminal%%" + this.id + "\n")
+
+  def assertSubscribed() : Unit = writer.write("assert%%subscribed%%" + this.id + "\n")
+
+  def assertNotSubscribed() : Unit = writer.write("assert%%no_subscribed%%" + this.id + "\n")
+
+  def assertCanceled() : Unit = writer.write("assert%%canceled%%" + this.id + "\n")
+
+  // await
+
+  def awaitTerminal() : Unit = writer.write("await%%terminal%%" + this.id + "\n")
+
+  def awaitOnNext() : Unit = writer.write("await%%onNext%%" + this.id + "\n")
+
+  def awaitAtLeast(n: Long, t : Long) = writer.write("await%%atLeast%%" + this.id + "%%" + n + "%%" + t + "\n")
+
+  def take(n: Long) : Unit = writer.write("take%%" + n + "%%" + this.id + "\n")
+
+  // internal functions
+
+  private def printList(lst: List[(String, String)]) : String = lst.map(a => a._1 + "," + a._2).mkString("&&")
+
+  private def writechannel(marble: Map[(String, String), String]) : String = {
+    marble.keys.map(a => a._1 + "@" + a._2 + ":" + marble.get(a).get).mkString(",")
+  }
+}
```

Comment:
So while I'm a *huge* Scala proponent (and also positively biased towards it, since at Lightbend/Akka), 
I'm not sure we need it here. I don't see the DSL actually make use of any Scala features (except the dot-less syntax, which is actually not really recommended for most apis). 

Would it be simpler to keep the TCK Java-only, as we did in Reactive Streams? (otherwise this library would have to be cross released for many scala versions, even for people using it only from Java).